batteries/argparse.luau

```diff
@@ -0,0 +1,137 @@
+local argparse = {}
+argparse.__index = argparse
+
+type ArgKind = "POSITIONAL" | "FLAG" | "OPTION"
```

Comment:
```suggestion
type ArgKind = "positional" | "flag" | "option"
```

I don't really see any reason for us to all-caps the kinds. Feels unidiomatic to me.